### PR TITLE
Change SW360 module to authenticate more general

### PR DESCRIPTION
**Pull request by blaumeiser-at-bosch**
Signed-off-by: Lars Geyer-Blaumeiser <lars.geyer-blaumeiser@bosch-si.com>

### CODE REVIEW

File: modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/rest/SW360AuthenticationClient.java [hunk 1]
```diff
@@ -26,14 +28,13 @@
 
 public class SW360AuthenticationClient {
     private final String GRANT_TYPE_VALUE = "password";
-    private final String LIFERAY_CLIENT_CREDENTIALS = "trusted-sw360-client:sw360-secret";
+//    private final String LIFERAY_CLIENT_CREDENTIALS = "trusted-sw360-client:sw360-secret";
```
Comment (maxhbr) [inline modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/rest/SW360AuthenticationClient.java]: commented code

File: modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/rest/SW360AuthenticationClient.java [hunk 2]
```diff
@@ -43,53 +44,41 @@ public SW360AuthenticationClient(String authServerUrl) {
...
-                }
-            } else {
-                throw new AntennaException("Could not request OAuth2 access token for [" + username + "].");
+        addBasicAuthentication(headers, clientId, clientPassword);
+        headers.add(HttpHeaders.CONTENT_TYPE, "application/x-www-form-urlencoded");
+
+        HttpEntity<String> httpEntity = new HttpEntity<>(body, headers);
+        Logger logger = LoggerFactory.getLogger(getClass());
+        logger.debug("URL: " + requestUrl);
+        logger.debug("Entity: " + httpEntity);
```
Comment (maxhbr) [inline modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/rest/SW360AuthenticationClient.java]: this logs the credentials (in debug mode)?

File: modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/rest/SW360AuthenticationClient.java (hunk 1 above)
Comment (blaumeiser-at-bosch) [reply modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/rest/SW360AuthenticationClient.java]: Ups, you are right, fixed with the latest update, result of beeing disturbed during work

File: modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/rest/SW360AuthenticationClient.java (hunk 2 above)
Comment (blaumeiser-at-bosch) [reply modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/rest/SW360AuthenticationClient.java]: Same applies here, also this has been removed.

**Comment by blaumeiser-at-bosch** [comment]
Adapted the example projects with an SW360 configuration

**Comment by maxhbr** [comment]
Also the assertions in the tests have to be adopted to the changes in the example projects.

**Comment by maxhbr** [comment]
FYI: https://github.com/eclipse/antenna/issues/185

**Pull request merged**